aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/BaseHandlerStd.java

```
@@ -1,163 +1,303 @@
+            final ProgressEvent<ResourceModel, CallbackContext> progress,
+            final List<DBClusterRole> roles
+    ) {
         final ResourceModel model = progress.getResourceModel();
         final CallbackContext callbackContext = progress.getCallbackContext();
-        for(final DBClusterRole dbClusterRole: Optional.ofNullable(roles).orElse(Collections.emptyList())) {
-            final ProgressEvent<ResourceModel, CallbackContext> progressEvent = proxy.initiate("rds::add-roles-to-dbcluster", proxyClient, model, callbackContext)
-                .translateToServiceRequest(modelRequest -> addRoleToDbClusterRequest(modelRequest.getDBClusterIdentifier(), dbClusterRole.getRoleArn(), dbClusterRole.getFeatureName()))
-                .makeServiceCall((modelRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(modelRequest, proxyInvocation.client()::addRoleToDBCluster))
-                .stabilize((addRoleToDbClusterRequest, addRoleToDBClusterResponse, proxyInvocation, modelRequest, context) ->
-                    isRoleStabilized(proxyInvocation, modelRequest, dbClusterRole, true))
-                .success();
+
+        for (final DBClusterRole dbClusterRole : Optional.ofNullable(roles).orElse(Collections.emptyList())) {
+            final ProgressEvent<ResourceModel, CallbackContext> progressEvent = proxy.initiate("rds::remove-roles-to-dbcluster", proxyClient, model, callbackContext)
+                    .translateToServiceRequest(modelRequest -> removeRoleFromDbClusterRequest(
+                            modelRequest.getDBClusterIdentifier(),
+                            dbClusterRole.getRoleArn(),
+                            dbClusterRole.getFeatureName()
+                    ))
+                    .makeServiceCall((modelRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(
+                            modelRequest,
+                            proxyInvocation.client()::removeRoleFromDBCluster
```

Same as AddRoleToDBCluster logic. This call can throw ` DBClusterRoleNotFound` handler can be stuck. Is it expected or should we ignore this exception?

---

Error ruleset called `REMOVE_ASSOC_ROLES_ERROR_RULE_SET` from `BaseHandler` explicitly ignores `DBClusterRoleNotFoundException`.